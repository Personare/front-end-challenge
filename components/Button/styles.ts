import styled from 'styled-components'
import media from 'styled-media-query'
import {ButtonProps} from './index'

export const Button = styled.button<ButtonProps>`
  align-self: center;
  background-color: rgb(205, 0, 107);
  color: #FFF;
  cursor: pointer;
  font-size: 2rem;
  width: 95%;
  border: none;
  padding: 1rem;
  margin: auto 5%;

  &:active {
    background-color: rgb(161, 35, 101);
  }

  &:hover {
    background-color: rgb(161, 35, 101);
  }

  ${media.greaterThan('medium')`
    width: 300px;
    font-size: 1.3rem;`
  }
`
