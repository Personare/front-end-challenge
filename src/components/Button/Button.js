import React from 'react'
import styled, { keyframes } from 'styled-components'
import {
  oneOfType, arrayOf, node, func, string
} from 'prop-types'

const Button = ({ children, onClick, type }) => (
  <CustomButton onClick={onClick} type={type}>
    {children}
  </CustomButton>
)

const zoomInOut = keyframes`
  from {
    transform: scale(0.8)
  }

  to {
    transform: scale(1)
  }
`

const CustomButton = styled.button`
    height: 100px;
    width: 100%;
    background-color: #3498DB;
    max-width: 100px;
    border-radius: 100%;
    border: none;
    position: fixed;
    bottom: 0;
    right: 0;
    margin: 20px;
    color:  #ffffff;
    animation: ${zoomInOut} 1s alternate infinite;
`

Button.propTypes = {
  children: oneOfType([arrayOf(node), node]).isRequired,
  onClick: func.isRequired,
  type: string
}


Button.defaultProps = {
  type: 'button'
}
export default Button
