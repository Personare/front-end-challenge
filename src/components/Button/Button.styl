.Btn
  transition: all .2s ease-in-out
  display: inline-block
  height: 2.25rem
  padding: 0 1.625rem
  margin: .375rem 0
  color: #fff
  background-color: transparent
  border: none
  border-radius: 2px
  text-transform: uppercase
  font-weight: 500
  cursor: pointer
  &-default
    background-color: #ff4081
    &:hover
      box-shadow: 0 0 4px rgba(0, 0, 0, .12), 1px 3px 4px rgba(0 ,0, 0, .2)
      background-color: lighten(#ff4081, 10%)
