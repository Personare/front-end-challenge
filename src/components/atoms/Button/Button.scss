.Button {
    border: none;
    border-radius: 2px;
    color: #fff;
    border: 2px solid #fff;
    background: none;
    cursor: pointer;
    display: inline-block;
    font-family: 'Roboto',arial,sans-serif;
    font-size: 16px;
    font-weight: 400;
    letter-spacing: .5px;
    line-height: 20px;
    min-height: 48px;
    min-width: 100px;
    padding: 15px 24px;
    text-align: center;
    text-decoration: none;
    text-transform: uppercase;
    will-change: background-color;
    position: relative;
    overflow: hidden;
    transition: color .5s cubic-bezier(0.19, 1, 0.22, 1);
    z-index: 1;

    &::before {
        content: '';
        display: block;
        width: 120%;
        height: 100%;
        background-color: #fff;
        position: absolute;
        top: 0;
        left: 0;
        transform-origin: 0% 100%;
        transform: rotate3d(0, 0, 1, -50deg) translate3d(0, -3em, 0);
        transition: all .4s cubic-bezier(0.19, 1, 0.22, 1);
        z-index: -1;
    }

    &:hover {
        color: #111;

        &::before {
            transform: rotate3d(0, 0, 0, 0);
        }
    }

    &.--disabled {
        border: 0;
        background-color: #b4b4b4;
        color: #757575;
        cursor: not-allowed;
        outline: 0;

        &::before {
            content: none;
        }

        &:hover {
            color: #757575;
        }
    }
}
