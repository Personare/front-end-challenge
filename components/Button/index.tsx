import React, {ButtonHTMLAttributes} from "react";
import * as S from './styles'

export type ButtonProps = {
    children: React.ReactNode,
} & ButtonHTMLAttributes<HTMLButtonElement>

const Button = ({children, ...props}:ButtonProps) =>(
  <S.Button {...props}>
      {children}
  </S.Button>
)


export default Button;
