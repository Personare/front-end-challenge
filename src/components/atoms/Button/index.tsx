import React from 'react';

import * as S from './styles';

type Props = {
  text: string;
  disabled?: boolean;
  onClick?: () => void;
};

const Button = ({ text, disabled, onClick }: Props) => {
  return (
    <S.Button disabled={disabled} onClick={onClick}>
      {text}
    </S.Button>
  );
};

export default Button;
