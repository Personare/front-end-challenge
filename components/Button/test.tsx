import { render, screen, fireEvent } from "@testing-library/react"
import Button from './'

const text = 'Testing button'

describe('components/Button', () => {
  it('Should match snapshot', () => {
    const { container } = render(<Button>{text}</Button>)
    expect(container).toMatchSnapshot()
  })

  it('Should call function onClick', () => {
    const onClick = jest.fn()
    render(<Button onClick={onClick}>{text}</Button>)

    fireEvent.click(screen.getByRole('button', { name: text }))

    expect(onClick).toBeCalled()
  })
})
