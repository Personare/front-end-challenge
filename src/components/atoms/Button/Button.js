import React from 'react';
import PropTypes from 'prop-types';
import './Button.scss';

const Button = ({ onClick, children }) => (
    <button onClick={onClick} className="Button">
        {children}
    </button>
);

Button.propTypes = {
    onClick: PropTypes.func,
    children: PropTypes.string
};

export default Button;
