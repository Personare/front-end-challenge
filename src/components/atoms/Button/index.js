import { connect } from 'react-redux';
import Button from './Button';

const mapStateToProps = store => ({
    isClosed: store.tarotReducer.isClosed
});

export default connect(mapStateToProps)(Button);
