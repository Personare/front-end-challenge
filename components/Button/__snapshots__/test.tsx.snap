// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button Should match snapshot 1`] = `
.c0 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  background-color: rgb(205,0,107);
  color: #FFF;
  cursor: pointer;
  font-size: 2rem;
  width: 95%;
  border: none;
  padding: 1rem;
  margin: auto 5%;
}

.c0:active {
  background-color: rgb(161,35,101);
}

.c0:hover {
  background-color: rgb(161,35,101);
}

@media (min-width:768px) {
  .c0 {
    width: 300px;
    font-size: 1.3rem;
  }
}

<div>
  <button
    class="c0"
  >
    Testing button
  </button>
</div>
`;
